docs/Linting.md

```diff
@@ -0,0 +1,103 @@
+| [eslint](https://eslint.org/)              |                                                    [JS/React/a11y best practise/style enforcement](#JS-code) |
+| [stylelint](https://stylelint.io/)         |                                               [Styling (css, scss) code style enforcement](#scss-/-css-code) |
+| [prettier](https://prettier.io/)           |                                                                       [Code format enforcement](#code-style) |
+
+For JS and SCSS/CSS code, the intent is to have a dedicated tool (ie `eslint`, `stylelint`) to enforce best practice/syntax rules, but both to delegate/integrate code formatting responsibilities to `prettier`.
+
+## Linting implementation and config
+
+All configuration and any implementation for linting in this codebase is available [here.](../utils/linting). This has been done deliberately to keep the configuration together, but also to not bloat the root directory of the repository with lots of configuration files. Do note however that some symlinks will be present in the root directory of the repository, as some of the linting tools used expect configuration to be present at the root level. These links will redirect to the `linting` folder in these cases. The configuration in the `linting` directory is deliberately modular to allow easy modification if required. These are then referenced and used in the top level `package.json` so they can be driven via npm scripts.
+
+## Linting rules
+
+This section will detail the rules implemented across the codebase, and a rationale to why these rules have been chosen. Apart from the [Git commit checks](#Git-commit-checks), all linting will occur pre commit, via a Husky hook ([`husky` config for this available here](../utils/linting/husky.config.js)). Alternatively, a developer can run these pre commit checks by running:
+
+```
+npm run lint
+```
+
+to run all checks, or for individual lint checks, the command mentioned in each section respectively.
+
+To pass linting, no errors or warnings are allowed - ie all rules must be satisfied. It is possible to override and ignore rules, but this should only be done as a last resort, with an accompanying explanation.
+
+### All files
+
+The following rules are applied to all files:
+
+| Rule/Ruleset                                                                              |                                       Rationale |                                                         Implemented by | Auto fixes? |
+| ----------------------------------------------------------------------------------------- | ----------------------------------------------: | ---------------------------------------------------------------------: | ----------: |
+| Must have the Strimzi copyright header (in the appropriate format) at the top of the file | Requirement for all code contributed to Strimzi | [license-check-and-add](https://github.com/awjh/license-check-and-add) |         Yes |
```

Comment:
where is the copyright header format defined? are you going to use the same we have in the Java source code?

Reply:
@samuel-hawker kindly provided me with this:
```
Copyright Strimzi authors.
License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
```
I would have this defined in a file, which the tooling then inserts as required (with the right comments)